src/openrct2-ui/windows/Guest.cpp

```diff
@@ -1308,7 +1308,9 @@ void window_guest_overview_tool_update(rct_window* w, rct_widgetindex widgetInde
         return;
     }
 
-    uint32_t imageId = g_peep_animation_entries[peep->SpriteType].sprite_animation[PEEP_ACTION_SPRITE_TYPE_UI].base_image;
+    uint32_t imageId = g_peep_animation_entries[peep->SpriteType]
```

Comment:
I was actually thinking more of a `GetAnimationEntry(SpriteType, ActionSpriteType)`. What do you think would that make sense?

Reply:
Well, as far as I could understand, I think that makes sense a function like `GetPeepAnimation(SpriteType, ActionSpriteType)`. Where it will be returned a `rct_peep_animation` given `SpriteType` and `ActionSpriteType`.

The function signature could be something like this: 
`const rct_peep_animation& GetPeepAnimation(const PeepSpriteType&, const PeepActionSpriteType&);`

The above line would look like this:
`uint32_t imageId = GetPeepAnimation(peep->SpriteType, PeepActionSpriteType::Ui).base_image;`